ocs_ci/ocs/ocs_upgrade.py

```diff
@@ -0,0 +1,363 @@
+        tuple: with three sets which contain those types of images
+            old_images_for_upgrade - images which should be upgraded
+            new_images_to_upgrade - new images to which upgrade
+            unchanged_images - unchanged images without upgrade
+
+    """
+    old_csv_images = set(old_csv_images.values())
+    new_csv_images = set(new_csv_images.values())
+    old_images_for_upgrade = old_csv_images - new_csv_images
+    log.info(
+        f"Old images which are going to be upgraded: "
+        f"{sorted(old_images_for_upgrade)}"
+    )
+    new_images_to_upgrade = new_csv_images - old_csv_images
+    log.info(
+        f"New images for upgrade: "
+        f"{sorted(new_images_to_upgrade)}"
+    )
+    unchanged_images = old_csv_images.intersection(new_csv_images)
+    log.info(
+        f"Unchanged images after upgrade: "
+        f"{sorted(unchanged_images)}"
+    )
+    return (
+        old_images_for_upgrade,
+        new_images_to_upgrade,
+        unchanged_images,
+    )
+
+
+def verify_image_versions(old_images, upgrade_version):
+    """
+    Verify if all the images of OCS objects got upgraded
+
+    Args:
+        old_images (set): set with old images
+        upgrade_version (packaging.version.Version): version of OCS
+
+    """
+    namespace = config.ENV_DATA['cluster_namespace']
+    number_of_worker_nodes = len(get_typed_nodes())
+    storage_cluster = StorageCluster(
+        resource_name=config.ENV_DATA['storage_cluster_name'],
+        namespace=namespace
+    )
+    osd_count = (
+        int(storage_cluster.data['spec']['storageDeviceSets'][0]['count'])
+        * int(storage_cluster.data['spec']['storageDeviceSets'][0]['replica'])
+    )
+    verify_pods_upgraded(old_images, selector=constants.OCS_OPERATOR_LABEL)
+    verify_pods_upgraded(old_images, selector=constants.OPERATOR_LABEL)
+    # in 4.3 app selector noobaa have those pods: noobaa-core-ID, noobaa-db-ID,
+    # noobaa-operator-ID but in 4.2 only 2: noobaa-core-ID, noobaa-operator-ID
+    noobaa_pods = 2 if upgrade_version < parse_version('4.3') else 3
+    verify_pods_upgraded(
+        old_images, selector=constants.NOOBAA_APP_LABEL, count=noobaa_pods
+    )
+    verify_pods_upgraded(
+        old_images, selector=constants.CSI_CEPHFSPLUGIN_LABEL,
+        count=number_of_worker_nodes,
+    )
+    verify_pods_upgraded(
+        old_images, selector=constants.CSI_CEPHFSPLUGIN_PROVISIONER_LABEL,
+        count=2
+    )
+    verify_pods_upgraded(
+        old_images, selector=constants.CSI_RBDPLUGIN_LABEL,
+        count=number_of_worker_nodes,
+    )
+    verify_pods_upgraded(
+        old_images, selector=constants.CSI_RBDPLUGIN_PROVISIONER_LABEL,
+        count=2
+    )
+    verify_pods_upgraded(old_images, selector=constants.MON_APP_LABEL, count=3)
+    verify_pods_upgraded(old_images, selector=constants.MGR_APP_LABEL)
+    verify_pods_upgraded(
+        old_images, selector=constants.OSD_APP_LABEL, count=osd_count,
+        timeout=1800,
```

Comment:
This timeout should be counted dynamically now based on OSD count, see here:  https://github.com/red-hat-storage/ocs-ci/commit/0374fbe43436018e2772feed1c188b250f175e8d#diff-40ca86338b56ecd56fb458ca2ea36c3b

Reply:
fixed